plugins/modules/mso_schema_site_vrf_region_cidr.py

```diff
@@ -187,79 +187,103 @@ def main():
-    if (site_id, template) not in sites:
-        mso.fail_json(msg="Provided site/siteId/template '{0}/{1}/{2}' does not exist. "
-                          "Existing siteIds/templates: {3}".format(site, site_id, template, ', '.join(sites_list)))
-
-    # Schema-access uses indexes
-    site_idx = sites.index((site_id, template))
-    # Path-based access uses site_id-template
-    site_template = '{0}-{1}'.format(site_id, template)
-
     payload = dict()
     op_path = ''
     new_cidr = dict(
         ip=cidr,
         primary=primary,
     )
 
+    # Get site
+    site_id = mso.lookup_site(site)
+
+    # Get site_idx
+    all_sites = schema_obj.get('sites')
+    sites = []
+    if all_sites is not None:
+        sites = [(s.get('siteId'), s.get('templateName')) for s in all_sites]
+
     # Get VRF
     vrf_ref = mso.vrf_ref(schema_id=schema_id, template=template, vrf=vrf)
-    vrfs = [v.get('vrfRef') for v in schema_obj.get('sites')[site_idx]['vrfs']]
     template_vrfs = [a.get('name') for a in schema_obj['templates'][template_idx]['vrfs']]
     if vrf not in template_vrfs:
         mso.fail_json(msg="Provided vrf '{0}' does not exist. Existing vrfs: {1}".format(vrf, ', '.join(template_vrfs)))
     else:
         # Update vrf index at template level
         template_vrf_idx = template_vrfs.index(vrf)
 
-    # If vrf not at site level but exists at template level
-    if vrf_ref not in vrfs:
-        op_path = '/sites/{0}/vrfs/-'.format(site_template)
+    # if site-template does not exist, create it
+    if (site_id, template) not in sites:
+        op_path = '/sites/-'
         payload.update(
-            vrfRef=dict(
-                schemaId=schema_id,
-                templateName=template,
-                vrfName=vrf,
-            ),
-            regions=[dict(
-                name=region,
-                cidrs=[new_cidr]
+            siteId=site_id,
+            templateName=template,
+            vrfs=[dict(
+                vrfRef=dict(
+                    schemaId=schema_id,
+                    templateName=template,
+                    vrfName=vrf,
+                ),
+                regions=[dict(
+                    name=region,
+                    cidrs=[new_cidr]
+                )]
             )]
         )
-    else:
-        # Update vrf index at site level
-        vrf_idx = vrfs.index(vrf_ref)
 
-        # Get Region
-        regions = [r.get('name') for r in schema_obj.get('sites')[site_idx]['vrfs'][vrf_idx]['regions']]
-        if region not in regions:
-            op_path = '/sites/{0}/vrfs/{1}/regions/-'.format(site_template, vrf)
+    else:
+        # Schema-access uses indexes
+        site_idx = sites.index((site_id, template))
+        # Path-based access uses site_id-template
+        site_template = '{0}-{1}'.format(site_id, template)
+
+        # If vrf not at site level but exists at template level
+        all_vrfs = schema_obj.get('sites')[site_idx]['vrfs']
```

Comment:
replace with vrfs = [v.get('vrfRef') for v in schema_obj.get('sites')[site_idx]['vrfs']]